userspace/libsinsp/sinsp.cpp

```diff
@@ -2795,6 +2820,27 @@ libsinsp::event_processor::build_threadinfo(sinsp* inspector)
 	return new sinsp_threadinfo(inspector);
 }
 
+void sinsp::handle_async_event(std::unique_ptr<sinsp_evt> evt)
+{
+	// see comments in handle_plugin_async_event
+	if(!is_capture())
+	{
+		evt->inspector(this);
+		if(evt->m_pevt->ts != (uint64_t)-1 &&
+		   evt->m_pevt->ts > sinsp_utils::get_current_time_ns() + ONE_SECOND_IN_NS * 10)
+		{
+			g_logger.log("async event ts too far in future", sinsp_logger::SEV_WARNING);
+			return;
+		}
+
+		if(!m_pending_state_evts.push(std::move(evt)))
+		{
+			g_logger.log("async event queue is full", sinsp_logger::SEV_WARNING);
+			throw sinsp_exception("async event queue is full");
```

Comment:
I'd avoid throwing an exception here; this is ideally the same as a driver event drop because of `PPM_BUFFER_FULL`.
I'd just log the warn and drop the event.

Reply:
The issue here would be for the plugin side: a plugin would never be able to tell wether the event has been rejected or not. This may be an obstacle in logics in which you send an async event and the "wait for it" in the parsing phase from the same plugin. WDYT? Better ideas?

Reply:
IMHO The plugin cannot rely on the fact that the event is surely sent, all the plugin logics should be written considering the fact that the event could be dropped... with a limited queue we cannot ensure that all events are always sent and terminate the whole inspector could be an excessive reaction, WDYT?

Reply:
Agreed!